### PR TITLE
Add consistent LRO operation fail behavior across Form Recognizer poller methods

### PR DESCRIPTION
**Pull request by samvaity**
Update poller methods to show consistent behavior when service returns model/result with failed/invalid status.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java [hunk 1]
```diff
@@ -82,16 +81,19 @@
...
      *
-     * @param fileSourceUrl The source URL to the input document. Size of the file must be less than 50 MB.
+     * @param formUrl The source URL to the input document. Size of the file must be less than 50 MB.
      * @param modelId The UUID string format custom trained model Id to be used.
      *
      * @return A {@link PollerFlux} that polls the extract custom form operation until it has completed, has failed,
      * or has been cancelled. The completed operation returns a List of {@link RecognizedForm}.
+     * @throws HttpResponseException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
+     * an {@link OperationStatus#FAILED}.
+     * @throws NullPointerException If {@code fileSourceUrl}, {@code modelId} is {@code null}.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:91]:
```suggestion
     * @throws NullPointerException If {@code formUrl}, {@code modelId} is {@code null}.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java [hunk 2]
```diff
@@ -103,23 +105,26 @@
...
      * @param modelId The UUID string format custom trained model Id to be used.
      * @param includeTextDetails Include text lines and element references in the result.
      * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
      * 5 seconds is used.
      *
      * @return A {@link PollerFlux} that polls the extract custom form operation until it has completed, has failed,
      * or has been cancelled. The completed operation returns a List of {@link RecognizedForm}.
+     * @throws HttpResponseException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
+     * an {@link OperationStatus#FAILED}.
+     * @throws NullPointerException If {@code fileSourceUrl}, {@code modelId} is {@code null}.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:118]:
```suggestion
     * @throws NullPointerException If {@code formUrl}, {@code modelId} is {@code null}.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java [hunk 3]
```diff
@@ -215,19 +229,22 @@
...
-     * @param sourceUrl The source URL to the input document. Size of the file must be less than 50 MB.
+     * @param formUrl The source URL to the input document. Size of the file must be less than 50 MB.
      * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
      * 5 seconds is used.
      *
      * @return A {@link PollerFlux} that polls the extract receipt operation until it has completed, has failed, or has
      * been cancelled. The completed operation returns a List of {@link FormPage}.
+     * @throws HttpResponseException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
+     * an {@link OperationStatus#FAILED}.
+     * @throws NullPointerException If {@code sourceUrl} is {@code null}.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:240]:
```suggestion
     * @throws NullPointerException If {@code formUrl} is {@code null}.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java [hunk 4]
```diff
@@ -301,15 +324,18 @@
      * <p><strong>Code sample</strong></p>
      * {@codesnippet com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeReceiptsFromUrl#string}
      *
-     * @param sourceUrl The source URL to the input document. Size of the file must be less than 50 MB.
+     * @param receiptUrl The source URL to the input document. Size of the file must be less than 50 MB.
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:327]: nit: "The source URL to the input receipt." 

Also unsure if size of the file is as necessary for the FromUrl functions

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # Release History
 
 ## 1.0.0-beta.3 (Unreleased)
+- Raise `HttpResponseException` when a model with `ModelStatus.Invalid` is returned from the `beginTraining()` API's
+- Fix `HttpResponseException` to include the error object thrown on invalid analyze status for recognize API's
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:5]: could we add the renames to the changelog too?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClient.java [hunk 6]
```diff
@@ -22,7 +25,7 @@
 
 /**
  * This class provides a synchronous client that contains all the operations that apply to Azure Form Recognizer.
- * Operations allowed by the client are recognizing receipt data from documents, extracting layout information and
+ * Operations allowed by the client are recognizing receipt data from documents, recognizeing layout information and
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClient.java:28]:
```suggestion
 * Operations allowed by the client are recognizing receipt data from documents, recognizing layout information and
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java [hunk 7]
```diff
@@ -378,37 +410,40 @@
...
      * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
      * 5 seconds is used.
      *
-     * @return A {@link PollerFlux} that polls the extract receipt operation until it has completed, has failed, or has
-     * been cancelled. The completed operation returns a List of {@link RecognizedReceipt}.
+     * @return A {@link PollerFlux} that polls the recognize receipt operation until it has completed, has failed,
+     * or has been cancelled. The completed operation returns a List of {@link RecognizedReceipt}.
+     * @throws HttpResponseException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
+     * an {@link OperationStatus#FAILED}.
+     * @throws NullPointerException If {@code form} is {@code null}.
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:424]:
```suggestion
     * @throws NullPointerException If {@code receipt} is {@code null}.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java [hunk 8]
```diff
@@ -638,4 +650,20 @@ private void throwIfCopyOperationStatusInvalid(CopyOperationResult copyResult) {
...
+      *  Helper method that throws a {@link HttpResponseException} if {@link ModelInfo#getStatus()} is
+      *  {@link com.azure.ai.formrecognizer.implementation.models.ModelStatus#INVALID}.
+      *
+      * @param customModel The response returned from the service.
+      */
+    private void throwIfModelStatusInvalid(Model customModel) {
+        if (ModelStatus.INVALID.equals(customModel.getModelInfo().getStatus())) {
+            List<ErrorInformation> errorInformationList = customModel.getTrainResult().getErrors();
+            if (!CoreUtils.isNullOrEmpty(errorInformationList)) {
+                throw logger.logExceptionAsError(new HttpResponseException(
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:663]: Should this be a `HttpResponseException` or just `AzureException`? `HttpResponseException` is usually an error returned by the HTTP call with a status code not equal to 2xx.

From [HttpResponseException JavaDoc](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/apidocs/com/azure/core/exception/HttpResponseException.html):
>The exception thrown when an unsuccessful response is received with http status code (e.g. 3XX, 4XX, 5XX) from the service request.

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:663]: For failures for service requests (3XX, 4XX etc) we have [ErroeResponseException](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/FormRecognizerClientImpl.java#L143).

Using the equivalent for other languages [`RequestFailedException`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/src/CopyModelOperation.cs#L163) and [`HttpResponseError`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py#L111)

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:663]: Discussed offline.
Maintaining the Javadoc for `HttpResponseException` and TA convention for exception handling, consider adding a new Exception Type or use AzureException.
https://github.com/Azure/azure-sdk-for-java/issues/11705

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:663]: https://github.com/Azure/azure-sdk-for-java/pull/11720

**Pull request merged**